src/main/java/org/apache/commons/compress/archivers/arj/ArjArchiveInputStream.java

```diff
@@ -99,15 +132,21 @@ public static boolean matches(final byte[] signature, final int length) {
         return length >= 2 && (0xff & signature[0]) == ARJ_MAGIC_1 && (0xff & signature[1]) == ARJ_MAGIC_2;
     }
 
-    private final DataInputStream dis;
+    private static int readUnsignedByte(InputStream in) throws IOException {
+        final int value = in.read();
+        if (value == -1) {
+            throw new EOFException();
```

Comment:
Add a message to the exception I think. It's nicer for both end users and developers IMO.

Reply:
Fixed in https://github.com/apache/commons-compress/pull/728/commits/ef7ef06b491573dd399c3cf25cfec3695445fbe6